eng/doc/DeveloperGuide.md

```diff
@@ -0,0 +1,84 @@
+# Developer Guide
+
+This document is a guide for developers who want to contribute to the Microsoft Go repository.
+It explains how to build the repository, how to work with the Go submodule, and how to use the different tools
+that help maintain the repository.
+
+## Setting up the repository
+
+### Step 0: Contributor License Agreement
+
+Most contributions require you to agree to a Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us the rights to use your contribution.
+For details, visit https://cla.opensource.microsoft.com.
+
+### Step 1: Install a Go toolchain
+
+A preexisting Go toolchain is required to bootstrap the build process.
+You can use your system's package manager to install Go, or you can download it from the [official Go website](https://golang.org/dl/).
+The only requirement is that the Go version is high enough for the bootstrap process.
+If the version is too low, the bootstrap process will fail and ask you to install a newer version.
+
+### Step 2: Install the git-go-patch command
```

Comment:
Maybe worth mentioning up front that `git` is required? (Step 1 and 2 could be combined to a `Step 1: Install prerequisites` to avoid adding a bunch of steps?)

Reply:
I like having the different prerequisites on their own sections 😸. Will mention `git`.